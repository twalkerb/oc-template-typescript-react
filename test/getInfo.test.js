const getInfo = require('../src/getInfo');

describe('when invoking getInfo()', () => {
  const info = getInfo();

  test('should return the correct version', () => {
    expect(info.version).toBeDefined();
  });
  test('should return the correct template type', () => {
    expect(info.type).toBe('oc-template-react');
  });
  test('should return the list of dependencies', () => {
    expect(info.dependencies).toMatchSnapshot();
  });
  test('should return the list of dependencies', () => {
    expect(info.externals).toMatchSnapshot();
  });
});
