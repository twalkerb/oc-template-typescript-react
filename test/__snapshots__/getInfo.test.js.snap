exports[`when invoking getInfo() should return the list of dependencies 1`] = `
Object {
  "babel-core": "^6.23.1",
  "babel-loader": "^6.3.2",
  "babel-preset-babili": "0.0.11",
  "babel-preset-es2015": "^6.22.0",
  "babel-preset-react": "^6.23.0",
  "babili-webpack-plugin": "0.0.10",
  "fs-extra": "^2.0.0",
  "memory-fs": "^0.4.1",
  "react": "^15.4.2",
  "react-dom": "^15.4.2",
  "uuid": "^3.0.1",
  "webpack": "^2.2.1",
}
`;

exports[`when invoking getInfo() should return the list of dependencies 2`] = `
Array [
  Object {
    "global": "React",
    "url": "https://cdnjs.cloudflare.com/ajax/libs/react/15.4.2/react.min.js",
  },
  Object {
    "global": "ReactDOM",
    "url": "https://cdnjs.cloudflare.com/ajax/libs/react/15.4.2/react-dom.min.js",
  },
]
`;
